src/modm/platform/can/sam-mcan/can_impl.hpp

```diff
@@ -0,0 +1,619 @@
+	}
+	else if (Regs()->MCAN_PSR & MCAN_PSR_EW) {
+		return BusState::ErrorWarning;
+	}
+	else {
+		return BusState::Connected;
+	}
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setStandardFilter(
+	uint8_t standardIndex, FilterConfig config,
+	modm::can::StandardIdentifier id_,
+	modm::can::StandardMask mask)
+{
+	if (standardIndex >= MessageRam::StandardFilterCount) {
+		return false;
+	}
+
+	MessageRam::setStandardFilter(standardIndex,
+		MessageRam::FilterType::Classic,
+		config, uint16_t(id_), uint16_t(mask));
+
+	return true;
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setStandardFilter(
+	uint8_t standardIndex, FilterConfig config,
+	modm::can::StandardIdentifier id0,
+	modm::can::StandardIdentifier id1)
+{
+	if (standardIndex >= MessageRam::StandardFilterCount) {
+		return false;
+	}
+
+	MessageRam::setStandardFilter(standardIndex,
+		MessageRam::FilterType::Dual,
+		config, uint16_t(id0), uint16_t(id1));
+
+	return true;
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setStandardRangeFilter(
+	uint8_t standardIndex, FilterConfig config,
+	modm::can::StandardIdentifier first,
+	modm::can::StandardIdentifier last)
+{
+	if (standardIndex >= MessageRam::StandardFilterCount) {
+		return false;
+	}
+
+	MessageRam::setStandardFilter(standardIndex,
+		MessageRam::FilterType::Range,
+		config, uint16_t(first), uint16_t(last));
+
+	return true;
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setExtendedFilter(
+	uint8_t extendedIndex, FilterConfig config,
+	modm::can::ExtendedIdentifier id_,
+	modm::can::ExtendedMask mask)
+{
+	if (extendedIndex >= MessageRam::ExtendedFilterCount) {
+		return false;
+	}
+
+	MessageRam::setExtendedFilter0(extendedIndex,
+		MessageRam::FilterConfig::Disabled, 0);
+	MessageRam::setExtendedFilter1(extendedIndex,
+		MessageRam::FilterType::Classic, uint32_t(mask));
+	MessageRam::setExtendedFilter0(extendedIndex,
+		config, uint32_t(id_));
+
+	return true;
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setExtendedFilter(
+	uint8_t extendedIndex, FilterConfig config,
+	modm::can::ExtendedIdentifier id0,
+	modm::can::ExtendedIdentifier id1)
+{
+	if (extendedIndex >= MessageRam::ExtendedFilterCount) {
+		return false;
+	}
+
+	MessageRam::setExtendedFilter0(extendedIndex,
+		MessageRam::FilterConfig::Disabled, 0);
+	MessageRam::setExtendedFilter1(extendedIndex,
+		MessageRam::FilterType::Dual, uint32_t(id1));
+	MessageRam::setExtendedFilter0(extendedIndex,
+		config, uint32_t(id0));
+
+	return true;
+}
+
+template<uint8_t id, fdcan::MessageRamConfig mrc>
+bool
+McanDriver<id, mrc>::setExtendedRangeFilter(
+	uint8_t extendedIndex, FilterConfig config,
+	modm::can::ExtendedIdentifier first,
+	modm::can::ExtendedIdentifier last)
+{
+	if (extendedIndex >= MessageRam::ExtendedFilterCount) {
+		return false;
+	}
+
+	EnterInitMode init;
```

Comment:
Init mode is set here, for other filter configuration functions it isn't. Is that intended?